hail/python/test/hailtop/aiotools/generate_copy_test_specs.py

```diff
@@ -0,0 +1,139 @@
+import secrets
+from concurrent.futures import ThreadPoolExecutor
+import pprint
+import asyncio
+from hailtop.aiotools import LocalAsyncFS, RouterAsyncFS, Transfer
+
+
+def remove_prefix(s, prefix):
+    assert s.startswith(prefix), (prefix, s)
+    return s[len(prefix):]
+
+
+async def create_test_file(fs, name, base, path):
+    assert not path.endswith('/')
+    async with await fs.create(f'{base}{path}') as f:
+        await f.write(f'{name}/{path}'.encode('utf-8'))
+
+
+async def create_test_dir(fs, name, base, path):
+    assert path.endswith('/')
+    await fs.mkdir(f'{base}{path}')
+    if name == 'src':
+        await create_test_file(fs, name, base, f'{path}file1')
+    await fs.mkdir(f'{base}{path}subdir/')
```

Comment:
Why does it matter whether the name is src or dest? Shouldn't the test dir always be the same?

Reply:
I made a clarifying comment.